libraries/persistent_store/fuzz/src/store.rs

```diff
@@ -0,0 +1,533 @@
+                println!("Start with {} consumed erase cycles.", cycle);
+            }
+            self.record(StatKey::InitCycles, cycle);
+            let storage = vec![0xff; storage_size].into_boxed_slice();
+            let mut storage = BufferStorage::new(storage, options);
+            Store::init_with_cycle(&mut storage, cycle);
+            StoreDriver::Off(StoreDriverOff::new_dirty(storage))
+        } else {
+            StoreDriver::Off(StoreDriverOff::new(options, num_pages))
+        }
+    }
+
+    /// Powers a driver with possible interruption.
+    fn power_on(&mut self, driver: StoreDriverOff) -> StoreDriver {
+        if self.debug {
+            println!("Power on the store.");
+        }
+        self.increment(StatKey::PowerOnCount);
+        let interruption = self.interruption(driver.delay_map());
+        match driver.partial_power_on(interruption) {
+            Err((storage, _)) if self.init.is_dirty() => {
+                self.entropy.consume_all();
+                StoreDriver::Off(StoreDriverOff::new_dirty(storage))
+            }
+            Err(error) => self.crash(error),
+            Ok(driver) => driver,
+        }
+    }
+
+    /// Generates and applies an operation with possible interruption.
+    fn apply(&mut self, driver: StoreDriverOn) -> Result<StoreDriver, Store<BufferStorage>> {
+        let operation = self.operation(&driver);
+        if self.debug {
+            println!("{:?}", operation);
+        }
+        let interruption = self.interruption(driver.delay_map(&operation));
+        match driver.partial_apply(operation, interruption) {
+            Err((store, _)) if self.init.is_dirty() => {
+                self.entropy.consume_all();
+                Err(store)
+            }
+            Err((store, StoreInvariant::NoLifetime)) => Err(store),
+            Err((store, error)) => self.crash((store.extract_storage(), error)),
+            Ok((error, driver)) => {
+                if self.debug {
+                    if let Some(error) = error {
+                        println!("{:?}", error);
+                    }
+                }
+                Ok(driver)
+            }
+        }
+    }
+
+    /// Reports a broken invariant and terminates fuzzing.
+    fn crash(&self, error: (BufferStorage, StoreInvariant)) -> ! {
+        let (storage, invariant) = error;
+        if self.debug {
+            print!("{}", storage);
+        }
+        panic!("{:?}", invariant);
+    }
+
+    /// Records a statistics if enabled.
+    fn record(&mut self, key: StatKey, value: usize) {
+        if let Some(stats) = &mut self.stats {
+            stats.add(key, value);
+        }
+    }
+
+    /// Increments a counter if statistics are enabled.
+    fn increment(&mut self, key: StatKey) {
+        if self.stats.is_some() {
+            *self.counters.get_mut(&key).unwrap() += 1;
+        }
+    }
+
+    /// Initializes all counters if statistics are enabled.
+    fn init_counters(&mut self) {
```

Comment:
This is called once after `new`. Why not just call it inside of `new` so users can't forget?